tests/unit-tests/lib/test_shmem.cpp

```diff
@@ -0,0 +1,72 @@
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2020 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef _WIN32
```

Comment:
Why this can't be tested on Windows too?

Reply:
I can't test it in windows. These are the errors that the GitHub actions shows: https://github.com/BOINC/boinc/runs/2107473806

```
(ClCompile target) -> 
         D:\a\boinc\boinc\lib\shmem.h(37,20): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [D:\a\boinc\boinc\win_build\unittests_vs2019.vcxproj]
         D:\a\boinc\boinc\lib\shmem.h(37,8): error C2146: syntax error: missing ';' before identifier 'create_shmem' [D:\a\boinc\boinc\win_build\unittests_vs2019.vcxproj]
         D:\a\boinc\boinc\lib\shmem.h(40,20): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [D:\a\boinc\boinc\win_build\unittests_vs2019.vcxproj]
         D:\a\boinc\boinc\lib\shmem.h(40,1): error C2086: 'int HANDLE': redefinition [D:\a\boinc\boinc\win_build\unittests_vs2019.vcxproj]
         D:\a\boinc\boinc\lib\shmem.h(40,8): error C2146: syntax error: missing ';' before identifier 'attach_shmem' [D:\a\boinc\boinc\win_build\unittests_vs2019.vcxproj]
         D:\a\boinc\boinc\lib\shmem.h(41,25): error C2146: syntax error: missing ')' before identifier 'hSharedMem' [D:\a\boinc\boinc\win_build\unittests_vs2019.vcxproj]
        ```

Reply:
Try to put this to the `include` section:

```
#if defined(_WIN32)
#include "boinc_win.h"
#endif
```